### PR TITLE
PTRENG-4192 - Refactors

### PR DESCRIPTION
**Pull request by chb0github**
Moved a few more shared functions and renamed them. Also created packages for various pieces

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -32,7 +40,7 @@ NEW FEATURES:
 
 NEW FEATURES:
 
-* Add `test.ExecuteTemplate()` and `util.UniversalPack()`
+* Add `test.ExecuteTemplate()` and `packer.Universal()`
```
Comment (alexhung) [inline CHANGELOG.md:43]: I don't think we need to retroactively rename the package here 😄

File: packer/packer.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
+package packer
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jfrog/terraform-provider-shared/predicate"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"reflect"
```
Comment (alexhung) [inline packer/packer.go:8]: I know `gofmt` doesn't care about the ordering here. May be not even `goimports` but we are pretty consistent ourselves putting 3rd party modules last.

**Review by alexhung — APPROVED** [review]
LGTM. I leave it up to you if you want to update based on my comments.

File: packer/packer.go (hunk 2 above)
Comment (chb0github) [reply packer/packer.go:8]: Oh, I just let goland do its thing. I didn't realize this was important

**Review by alexhung — APPROVED** [review]

**Pull request merged**